### ISSUE TITLE
Problem handling start action when reloading configs 

### ISSUE THREAD

**Issue by jungwon-w** [strongswan/strongswan#2418]
**System (please complete the following information):**
 - OS: Rocky Linux  8.10
 - Kernel version (if applicable): 5.10
 - strongSwan version(s): last master version (2024-08-22)
 - Tested/confirmed with the latest version: yes

**Describe the bug**

If the host acts as the initiator(start_action = start), reloading the configs(swanctl --load-conns) while the connection is still being attempted(connection is not established yet) will cause the IKE_SA to close.

This issue does not occur in version 5.9.14.

Merge branch 'vici-reload-actions' (commit : 72f9a21b22e9807a258f2a982a3ce9cfc8b48ed1) - This appears to be caused by this patch.

**To Reproduce**
1. config :

```
gw-1 {
 local_addrs = 10.0.10.2
 remote_addrs = 10.0.10.1
 keyingtries = 0
 if_id_in = 1
 if_id_out = 1
 encap = no
 unique = no

 local {
  auth = psk
 }
 remote {
  auth = psk
 }
 
 children {
  child {
    local_ts = 0.0.0.0/0
    remote_ts = 0.0.0.0/0

    start_action = start
    close_action = start
    dpd_action = restart
  }
 }
}

```
2. Start 'gw-1' connection 
3. Run the 'swanctl --load-conns' command while attempting to connect.
4. Closing IKE_SA 'gw-1' and no further connection attempts will be made.

**Expected behavior**

**Logs/Backtraces**
Command execution result :

```
# swanctl --load-conns
loaded connection 'gw-1'
successfully loaded 1 connections, 0 unloaded
```

Logs :

```
2024-08-23T15:46:29.828300+09:00 ns info daemon ipsec:  05[CFG] added vici connection: gw-1
2024-08-23T15:46:29.828325+09:00 ns info daemon ipsec:  05[CFG] vici initiating 'child'
2024-08-23T15:46:29.828528+09:00 ns notice daemon ipsec:  05[IKE] <gw-1|13> initiating IKE_SA gw-1[13] to 10.0.10.1
2024-08-23T15:46:29.829200+09:00 ns info daemon ipsec:  05[ENC] <gw-1|13> generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) V ]
2024-08-23T15:46:29.829442+09:00 ns info daemon ipsec:  05[NET] <gw-1|13> sending packet: from 10.0.10.2[500] to 10.0.10.1[500] (1100 bytes)
2024-08-23T15:46:33.829694+09:00 ns info daemon ipsec:  06[IKE] <gw-1|13> retransmit 1 of request with message ID 0
2024-08-23T15:46:33.829757+09:00 ns info daemon ipsec:  06[NET] <gw-1|13> sending packet: from 10.0.10.2[500] to 10.0.10.1[500] (1100 bytes)
2024-08-23T15:46:41.029960+09:00 ns info daemon ipsec:  14[IKE] <gw-1|13> retransmit 2 of request with message ID 0
2024-08-23T15:46:41.030020+09:00 ns info daemon ipsec:  14[NET] <gw-1|13> sending packet: from 10.0.10.2[500] to 10.0.10.1[500] (1100 bytes)
2024-08-23T15:46:53.990237+09:00 ns info daemon ipsec:  15[IKE] <gw-1|13> retransmit 3 of request with message ID 0
2024-08-23T15:46:53.990299+09:00 ns info daemon ipsec:  15[NET] <gw-1|13> sending packet: from 10.0.10.2[500] to 10.0.10.1[500] (1100 bytes)
2024-08-23T15:46:54.664127+09:00 ns info daemon ipsec:  12[CFG] updated vici connection: gw-1
2024-08-23T15:46:54.664182+09:00 ns info daemon ipsec:  12[CFG] vici closing IKE_SA 'gw-1' #13
2024-08-23T15:46:54.664204+09:00 ns info daemon ipsec:  12[IKE] <gw-1|13> destroying IKE_SA in state CONNECTING without notification
```

**Additional context**

**Comment by tobiasbrunner** [comment]
Thanks for the report.

I was aware of potential issues with queued child-creating tasks, however, I didn't consider IKE_SAs that are not yet established and have therefore only such tasks queued but no CHILD_SAs.

A simple fix would be to just check for _any_ child-creating tasks and consider such IKE_SAs not empty, so they won't get deleted. However, without looking at the actual tasks, there could be some additional unresolved corner cases. A queued task might have to actually get removed if it's for a modified connection, and if the task is already active (e.g. for the first CHILD_SA of a new IKE_SA), we might have to delete the created CHILD_SA right away.

I've tried to implement that in the _2418-vici-reload-tasks_ branch. Would be great if you could test it.

**Comment by jungwon-w** [comment]
Thank you for your efforts.

I tested it with the 2418-vici-reload-tasks branch version and the above mentioned issue did not occur.
However, another problem was identified.
I'm not sure if this issue is caused by the related patch or a different patch, so I've added it as a new issue. (Incoming connections are established to IP addresses not specified in <conn>.local_addrs #2441)
The new issue does not occur in version 5.9.14, but in the latest master version and the 2418-vici-reload-tasks branch.
